bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/internal/BlueGigaReschedulableTimer.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.bluegiga.internal;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Simple reschedulable Timer.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+@NonNullByDefault
+public class BlueGigaReschedulableTimer extends Timer {
```

Comment:
Why do you need a custom timer implementation? Also you are not specifying the name of the thread the timer uses (if it can be specified) so when profiling/debugging it wouldn't be clear the binding responsible for it.

Reply:
It make easier to re schedule a timer as you can just scheduled again with new delay but it's still executed only ones. It's used to active passive scan after idle delay after executed command.

Not sure what you mean by the thread of the timer? Timer is named, but it doesn't contain information about BlueGiga but just general name "PassiveScanIdleTimer". Will add information about BlueGiga to name.

Reply:
Sorry about that, I don't use Timer enough to recognize that you named it. Looks like you fixed my concern anyway though. :)